src/azure/attestation/test/attestation_test.cc

```diff
@@ -134,4 +136,64 @@ TEST_F(JsonAttestationReportTest, FetchRealAttestationNonSnp) {
   }
   EXPECT_FALSE(fetchSnpAttestation().has_value());
 }
+
+TEST_F(JsonAttestationReportTest, GetReport) {
+  if (!hasSnp()) {
+    return;
+  }
+  // Define test inputs and outputs
+  std::string report_data = "";
+  uint8_t expected[64] = {
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+      0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+  };
+
+  std::unique_ptr<SnpReport> snp_report = getReport(report_data);
+
+  EXPECT_TRUE(std::equal(std::begin(expected), std::end(expected),
```

Comment:
std::array operator == should manage this such that EXPECT_EQ(expected_report, resulting_report) is good, conside and make sense quickly. It also avoids having to thing about the scope of the result of the get() on the unique ptr.

Reply:
If fact you seem to have done that at the top

Reply:
Yes I am pro using std::array I only chose uint8_t here because I thought you wanted minimal transformations or helper code so I'm happy to make that change

Reply:
https://github.com/KenGordon/data-plane-shared-libraries/pull/49/commits/c747c834323474ad2c624c92470f40f59f729099